src/coreclr/classlibnative/bcltype/objectnative.cpp

```diff
@@ -65,6 +65,34 @@ FCIMPL1(Object*, ObjectNative::GetObjectValue, Object* obj)
 }
 FCIMPLEND
 
+static INT32 TryGetHashCodeHelper(OBJECTREF objRef)
```

Comment:
This should be marked FORCEINLINE. I do not think that all C++ compiler is going to inline this method on its own that would cause performance regression.

Reply:
I reverted the changes to the existing GetHashCode function.